PennMobile/New Group/PAC Code/PacCodeViewController.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  PacCodeViewController.swift
+//  PennMobile
+//
+//  Created by CHOI Jongmin on 5/3/2020.
+//  Copyright © 2020 PennLabs. All rights reserved.
+//
+
+import Foundation
+
+class PacCodeViewController : UIViewController, KeychainAccessible, ShowsAlert, IndicatorEnabled {
+    
+    var pacCode : String?
+        
+    lazy var quadDigitLabel = [digitLabel, digitLabel, digitLabel, digitLabel]
+    
+    let pacCodeHStack = UIStackView()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .white
+        
+        self.title = "PAC Code"
+        
+        self.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .refresh, target: self, action: #selector(refreshButtonTapped))
+        
+        setupPacCodeHStack()
+        setupPacCodeInfoLabel()
+    }
+    
+    var digitLabel : UILabel {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.heightAnchor.constraint(equalToConstant: 70).isActive = true
+        label.widthAnchor.constraint(equalToConstant: 70).isActive = true
+        
+        label.textAlignment = .center
+        label.font = UIFont.boldSystemFont(ofSize: 30)
+        
+        label.backgroundColor = .grey6
+        
+        return label
+    }
+
+    func setupPacCodeHStack() {
+        pacCodeHStack.axis = .horizontal
+        pacCodeHStack.distribution = .equalCentering
+        pacCodeHStack.spacing = 10.0
+        
+        for digitLabel in quadDigitLabel {
+            pacCodeHStack.addArrangedSubview(digitLabel)
+        }
+        
+        view.addSubview(pacCodeHStack)
+        pacCodeHStack.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeHStack.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor).isActive = true
+        pacCodeHStack.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func setupPacCodeInfoLabel() {
+        let pacCodeInfoLabel = UILabel()
+        pacCodeInfoLabel.textColor = .labelPrimary
+        pacCodeInfoLabel.font = UIFont.systemFont(ofSize: 35)
+        pacCodeInfoLabel.text = "Your PAC Code is"
+        
+        view.addSubview(pacCodeInfoLabel)
+        pacCodeInfoLabel.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeInfoLabel.bottomAnchor.constraint(equalTo: pacCodeHStack.topAnchor, constant: -30).isActive = true
+        pacCodeInfoLabel.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func refreshPacCode() {
```

Comment:
This function is a bit of a misnomer. When I read `refreshPacCode`, I imagine that we're actually refreshing it from the CampusExpress. Let's change it to something else, like `setPACCode` or `updatePACCode`.